src/modules/init_maps.rs

```diff
@@ -425,3 +430,42 @@ fn map_events(blk: eth::Block) -> Result<Events, Error> {
 
     Ok(events)
 }
+
+#[substreams::handlers::map]
+fn map_indexing_rewards(storage_changes: StorageChanges, events: Events) -> Result<IndexingRewards, Error> {
+    let mut indexing_rewards = IndexingRewards::default(); 
+    let mut indexing_rewards_vec = vec![];
+    let delegation_pools = storage_changes.delegation_pools.unwrap();
+    let rewards_assigned_events = events.rewards_assigned_events.unwrap();
+    for rewards_assigned in rewards_assigned_events.rewards_assigned_events {
+        indexing_rewards_vec.push(IndexingReward {
+            id: rewards_assigned.id.clone(), // Each event needs a unique id
+            indexer: rewards_assigned.clone().indexer,
+            amount: rewards_assigned.clone().amount, // Tokens is origanally BigInt but proto does not have BigInt so we use string
+            indexer_rewards: rewards_assigned.amount.to_string(),
+            delegator_rewards: "0".to_string(),
+            ordinal: rewards_assigned.ordinal,
+        });
+        for delegation_pool in &delegation_pools.delegation_pools {
+            if delegation_pool.rewards == "true" {
+            log::info!("sth{:?}{:?}", rewards_assigned.indexer,delegation_pool.indexer );
+            if rewards_assigned.clone().indexer == delegation_pool.indexer && rewards_assigned.ordinal == delegation_pool.ordinal {
+                log::info!("stj", );
+                let delegator_rewards = BigInt::from_str(&delegation_pool.new_stake).unwrap().sub(BigInt::from_str(&delegation_pool.old_stake).unwrap());
+                let indexer_rewards = BigInt::from_str(&rewards_assigned.clone().amount).unwrap().sub(delegator_rewards.clone());
+                indexing_rewards_vec.push(IndexingReward {
+                    id: rewards_assigned.id.clone(), // Each event needs a unique id
+                    indexer: rewards_assigned.clone().indexer,
+                    amount: rewards_assigned.clone().amount, // Tokens is origanally BigInt but proto does not have BigInt so we use string
+                    indexer_rewards: indexer_rewards.to_string(),
+                    delegator_rewards: delegator_rewards.clone().to_string(),
+                    ordinal: rewards_assigned.ordinal,
+                });
```

Comment:
I might be missing something, but on these cases, aren't we going to have another `IndexingReward` message with the same ID as the one before (and all of the times it repeats inside the for loop), with also the same ordinal, but potentially different values?
Is that expected?

Reply:
Hey Juan, there is a logic I am trying to implement but it is not currently working, might have to refactor it. But currently, with that code I am trying to override the IndexingReward message with nonzero delegator_rewards if there are delegator rewards. The default of that field is zero.